arch/arm/core/aarch32/cortex_m/fpu.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2019,2020 Linaro Limited
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <arch/arm/aarch32/cortex_m/cmsis.h>
+
+/**
+ * @file @brief Helper functions for saving and restoring the FP context.
+ *
+ */
+
+#define CALLEE_SAVED_OFFS 16
+#define FPSCR_OFFS 32
+#define BUF_SIZE (FPSCR_OFFS + 1)
+
+int z_arm_save_fp_context(uint32_t *buffer, size_t buffer_size_words, bool *context_saved)
```

Comment:
I wonder if it would be better to define a struct for "FP context" and use that here instead. Also, it would be nice to have some integration with the `__esf` struct, in that regard:

https://github.com/zephyrproject-rtos/zephyr/blob/21d1ad3762302b3e461953df59430c77e0709274/include/arch/arm/aarch32/exc.h#L85

Reply:
Of it includes callee saved, not sure if it makes a lot of sense integrate with the exception stack frame...

Reply:
Made a stuct